providers/config.go

```diff
@@ -0,0 +1,100 @@
+package providers
+
+import (
+	"errors"
+	"fmt"
+	"html/template"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	// configFileLoc is the name of the config file.
+	configFileRelativeLoc = "providers.yaml"
+)
+
+var (
+	ErrProviderConfigNotFound = errors.New("provider config not found")
+)
+
+// Config is the entire configuration for all providers.
+type Config struct {
```

Comment:
Can we name this something other than Config is it doesn't get confused with the `Config` in our backend (which means the configuration of an installation). Maybe `Catalog`?

Reply:
Renamed this type, but let me know if you want me to avoid the word 'Config' even in other symbols like ReadConfig / ProviderConfig, etc.